File: src/PavementCondition.DataAccess/DatabaseContext.fs
namespace PavementCondition.DataAccess

open Microsoft.EntityFrameworkCore
open PavementCondition.Entity
open System.Reflection

type DatabaseContext(options: DbContextOptions<DatabaseContext>) =
    inherit DbContext(options)

    [<DefaultValue>]
    val mutable users : DbSet<User>

    member public this.Users with get() = this.users
                               and set p = this.users <- p
    
    override __.OnModelCreating(modelBuilder : ModelBuilder) =
        base.OnModelCreating(modelBuilder);
        modelBuilder.ApplyConfigurationsFromAssembly(Assembly.GetAssembly(typeof<DatabaseContext>))
        |> ignore

File: src/PavementCondition.DataAccess/DatabaseContext.cs
using Microsoft.EntityFrameworkCore;

using PavementCondition.Entity;

namespace PavementCondition.DataAccess
{
    public class DatabaseContext : DbContext
    {
        public DatabaseContext(DbContextOptions<DatabaseContext> options)
          : base(options)
        {
        }

        public DbSet<User> Users { get; set; }

        public DbSet<DefectType> DefectTypes { get; set; }

        public DbSet<Road> Roads { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
            modelBuilder.ApplyConfigurationsFromAssembly(this.GetType().Assembly);
        }
    }
}
